nengo_dl/converter.py

```diff
@@ -84,6 +84,11 @@ class Converter:
         increasing the synaptic filtering will make the network less responsive to
         rapid changes in the input, and you may need to present each input value
         for more timesteps in order to allow the network output to settle.
+    temporal_model : bool
+        Set to True if the Keras model contains a temporal
+        dimension (e.g. the inputs and outputs of each layer have shape
+        ``(batch_size, n_steps, ...)``). Note that all layers must be temporal, the
+        Converter cannot handle models with a mix of temporal and non-temporal layers.
```

Comment:
I'm trying to think of ways we could get around this for at least some of the simple/more common cases. It's pretty common for models to collapse over time in some way near the end.

What if rather than removing the first two dimensions for all layers, we only do that if `rank > 2`. So any layer that's already flattened would be treated as not having time. That would at least allow e.g. a dense layer at the end of the model that does the classification.

Reply:
Never mind. While the shape trick is possible, there's no way to pool across time in Nengo, since the whole network has to run each timestep.

Probably the best way to do something equivalent is to have time in all Keras layers, but then have a filter near the end of the network, and a loss function that just uses the last timestep. We could automate this "use the last timestep in the loss function" if e.g. we detect a filter layer with `return_sequences=False`, but it's not as trivial as I thought (and probably somewhat fragile, too).